### PR TITLE
refactor(svelte): extract GGPlot paint readiness and interaction pure seams

### PR DESCRIPTION
**Pull request by ljodea**
## Summary

- Extract canvas first-paint readiness into `plot-paint.ts` (`createPaintLedger`, `isPlotReady`) and derive `[data-gg-ready]` instead of effect-syncing it.
- Fix paint tracking so readiness requires each **distinct** canvas stratum of the current model run (re-attachment of one canvas can no longer satisfy the count alone).
- Move tool filtering, traversal hit building, interval selection assembly, and clear-inspection fingerprint ownership into focused pure modules with characterization unit tests.
- Keep `GGPlot` as the Svelte orchestrator; public API (`GGPlot`, `resetScales`, `setZoom`, props/events) is unchanged.

## Why

`GGPlot.svelte` still mixes pipeline/compositing readiness with interaction pure assembly after earlier helper extracts. These seams have clear unit-test surfaces and fix a real multi-canvas readiness edge case without touching the gesture state machine wholesale (a larger interval-interaction host extract remains a follow-up).

## Test plan

- [x] Unit tests: `plot-paint`, `plot-capability` (filterAvailableTools), `plot-interval` (lineage + intervalSelectionFromRows), `plot-pointer` (buildTraversalHits), clearInspectionFingerprint
- [x] Full `packages/svelte` browser suite (761 passed; one webkit hydration timeout flaked once and passed on re-run)
- [x] SSR harness (3 passed)
- [x] `bun run check:svelte`
- [x] `bun run lint:type-aware`
- [x] `bun run knip`
- [x] Independent Codex pre-PR review: no P1/P2 findings

### CODE REVIEW

**Comment by ljodea** [comment]
Follow-up for unrebutted post-merge/pre-merge-async findings: #64.

### Ledger
- **Valid → fix PR:** `data-gg-ready` via `$derived` emitted `true` during SSR for fixed-width SVG plots; decision 0009 requires `$effect` so prerender stays not-ready until the first client committed flush.

File: packages/svelte/src/lib/GGPlot.svelte [hunk 1]
```diff
@@ -1629,16 +1630,19 @@
...
-      (!hasCanvas ||
-        (paintedFor === model.runId && paintedCount >= canvasCount));
+  // `[data-gg-ready="true"]`). Derived after flush-visible state updates;
+  // canvas strata additionally gate on first paint (decision 0006 / plan).
+  const ready = $derived.by(() => {
+    void paintEpoch;
+    return isPlotReady({
+      hasModel: model !== null,
+      widthMode:
+        width === undefined || width === "container" ? "container" : "fixed",
```
[the remark this answers is not in the document]
Comment (ljodea) [reply packages/svelte/src/lib/GGPlot.svelte:1640]: ### ✅ Fixed in follow-up #64

Restored `$state(false)` + `$effect` so SSR stays `data-gg-ready="false"`; readiness still uses pure `isPlotReady` after the first client flush.
